cartridges/plugin_beam/cartridge/templates/default/beam/beam_select_nonprofit.isml

```diff
@@ -33,6 +34,57 @@
 
     initNetworkListeners(); // Will emit Beam network call events whenever the site makes an AJAX call
 
+    async function updateCustomAttributes(payload) {
+        try {
+            const url = `/on/demandware.store/Sites-RefArch-Site/${currLocale}/Cart-UpdateBeamCustomAttributes`
+            const response = await fetch(url, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(payload)
+            });
+            // Check for HTTP errors
+            if (!response.ok) {
+                console.error('Network response was not ok:', response.statusText);
+                return;
+            }
+
+            const result = await response.json();
+
+            if (!result.success) {
+                console.error('Error updating basket custom attributes:', result.error);
+            }
+        } catch (error) {
+            console.error('Fetch error:', error);
+        }
+    }
+
+    function getBeamCartId() {
+        // Get the beamCartId from localStorage
+        const beamCartIdKey = `beam_cart_key_${sitePref.beamApiKey}`;
+        const beamCartObj = window.localStorage.getItem(beamCartIdKey);
```

Comment:
`request` Does the SDK expose a method to get the Beam Cart Id out of localstorage || cookies? Seems like this method probably does what we'd need, but it'd be nice to not tie this custom code to a `beam_cart_key_` format that could change under the hood.

Reply:
ahh I think we actually may have access to `createScopedLocalStorage` let me try it out

Reply:
It turns out we don't currently have access to createScopedLocalStorage or getCartId. I'll be exporting those in a follow-up ticket. In the meantime, we do have access to getBeamCartCookie, so I'm using that for now and implementing the local storage logic as a fallback. @jmastic-beam what do you think?